src/controller/python/ChipDeviceController-ScriptBinding.cpp

```diff
@@ -214,6 +215,9 @@ ChipError::StorageType pychip_DeviceController_NewDeviceController(chip::Control
     (*outDevCtrl)->SetUdpListenPort(0);
     err = (*outDevCtrl)->Init(initParams);
     VerifyOrReturnError(err == CHIP_NO_ERROR, err.AsInteger());
+#if CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY
```

Comment:
Will this require separate compile options to have the python build in one mode vs the other? How does this work? 

I have a hard time matching PR description with change contents.

Reply:
Because we have these flags in the codebase anyway, we're already de-facto compiling with different flags. They're plumbed through gn normally, but we can compile with ex. different mdns impls etc. But I don't actually think this needs to be Ifdef'd - it seems like you can call StartServer and it will start advertising as a commissioner if that flag is on.

However, I agree with Andrei - it's unclear how you would compile the python controller to enable or disable this functionality.

Reply:
Why is the python controller required here?

Reply:
Filed this:
https://github.com/project-chip/connectedhomeip/issues/9743